qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -442,84 +581,115 @@ def __init__(self):
         super().__init__()
         self._decomposer_cache = {}
 
-    def _find_decomposer_2q_from_target(self, target, qubits, pulse_optimize):
-        qubits_tuple = tuple(qubits)
-        reverse_tuple = (qubits[1], qubits[0])
+    def _decomposer_2q_from_target(self, target, qubits, approximation_degree):
+        # we just need 2-qubit decomposers, in any direction.
+        # we'll fix the synthesis direction later.
+        qubits_tuple = tuple(sorted(qubits))
```

Comment:
Do you rely on this sorting or is it just for convenience? It seems, for instance, self._decomposer_cache is no longer direction sensitive?

Reply:
yeah it is not direction sensitive. I always sort the qubits before inserting in the cache and reading from the cache. If there's a direction mismatch it'll be fixed later.

Reply:
ok